src/Cardano/Shell/Constants/Types.hs

```diff
@@ -0,0 +1,276 @@
+{-# LANGUAGE DeriveAnyClass #-}
+
+module Cardano.Shell.Constants.Types
+    ( CardanoConfiguration (..)
+    , Core (..)
+    , Genesis (..)
+    , Spec (..)
+    , Initializer (..)
+    , TestBalance (..)
+    , FakeAvvmBalance (..)
+    , BlockVersionData (..)
+    , LastKnownBlockVersion (..)
+    , SoftForkRule (..)
+    , TxFeePolicy (..)
+    , TxSizeLinear (..)
+    , ProtocolConstants (..)
+    , NTP (..)
+    , Update (..)
+    , TXP (..)
+    , SSC (..)
+    , DLG (..)
+    , Block (..)
+    , Node (..)
+    , TLS (..)
+    , Wallet (..)
+    , Throttle (..)
+    , Certificate (..)
+    ) where
+
+import           Cardano.Prelude
+
+--------------------------------------------------------------------------------
+-- Cardano Configuration Data Structures
+--------------------------------------------------------------------------------
+-- | The basic configuration structure. It should contain all the required
+-- configuration parameters for the modules to work.
+
+data CardanoConfiguration = CardanoConfiguration
+    { ccLogPath             :: !FilePath
+    -- ^ The location of the log files on the filesystem.
+    , ccLogConfig           :: !FilePath
+    -- ^ The location of the log configuration on the filesystem.
+    , ccDBPath              :: !FilePath
+    -- ^ The location of the DB on the filesystem.
+    , ccApplicationLockFile :: !FilePath
+    -- ^ The location of the application lock file that is used
+    -- as a semaphore se we can run just one application
+    -- instance at a time.
+    , ccCore                :: !Core
+    , ccNTP                 :: !NTP
+    , ccUpdate              :: !Update
+    , ccTXP                 :: !TXP
+    , ccSSC                 :: !SSC
+    , ccDLG                 :: !DLG
+    , ccBlock               :: !Block
+    , ccNode                :: !Node
+    , ccTLS                 :: !TLS
+    , ccWallet              :: !Wallet
+    } deriving (Eq, Show, Monoid, Semigroup)
```

Comment:
Do you need the `Monoid` and `Semigroup`?

Reply:
Removed, thank you for pointing that out.